### PR TITLE
Add pre filter and shard counting step

### PR DESCRIPTION
**Pull request by mashehu**

### CODE REVIEW

File: .github/actions/number-of-shards/action.yml
Comment (edmundmiller) [inline .github/actions/number-of-shards/action.yml:1]: I love that you found that we can do composite actions in the same repo ❤️

**Review by maxulysse — APPROVED** [review]

File: .github/actions/number-of-shards/action.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    - name: Install nf-test
+      uses: nf-core/setup-nf-test@v1
+      with:
+        version: ${{ env.NFT_VER }}
+
+    - name: Get number of shards
+      shell: bash
+      run: |
+        nftest_output=$(nf-test test --dry-run --changed-since HEAD^ --filter process --follow-dependencies)
+        if nftest_output | grep -q 'No tests found'; then
```
Comment (mashehu) [inline .github/actions/number-of-shards/action.yml:26]:
```suggestion
        if $nftest_output | grep -q 'No tests found'; then
```

**Comment by mashehu** [comment]
I wouldn't at special paths just all modules and subworkflows. Then the filter step will take care of only picking up GPU tests

**Comment by sateeshperi** [comment]
why aren't some tests not launching ?

**Comment by mashehu** [comment]
they are not stalled, they are just required, but never launched (because we don't have anything to test). so we need to remove them from the ruleset and add a step which confirms the passing of all tests (either that they ran successfully or that they didn't have to be run)

**Comment by sateeshperi** [comment]
closing this as we got some of the logic in and will polish in another PR with working fixes from methylseq

**Pull request closed without merging**